Sources/SwiftProtobuf/Google_Protobuf_ListValue+Extensions.swift

```diff
@@ -43,15 +43,18 @@ extension Google_Protobuf_ListValue: _CustomJSONCodable {
     if decoder.scanner.skipOptionalNull() {
       return
     }
-    try decoder.scanner.skipRequiredArrayStart()
-    if decoder.scanner.skipOptionalArrayEnd() {
+    // Unlike most lists, this counts as a object start (for recursion depth);
+    // this follows protostream_objectsource.cc's
+    // ProtoStreamObjectSource::RenderField() logic for tracking the depth.
+    try decoder.scanner.skipRequiredArrayObjectStart()
```

Comment:
So if we count every array as an object, then we cut in half the depth limit when using repeated message fields.  i.e. - the array for the field's value is +1, and then each message parsed is +1 for the _{...}_ block.

```json
{
  "repeatedMessages": [
    { "anotherRepeatedMessage": [
         { "a": 1 }
       ]
    },
  ]
}
```

Currently this counts as depth 3, but if we count every list, it would be 5.

Reply:
I think it should count as 5.  The point here is to provide a sanity check on any input that risks excessive stack usage:  Arrays `[]` and Objects `{}` can both result in recursive function calls to process their contents, so I think it makes sense to count them both as part of that check.

Reply:
Should we change our default depth (for JSON) then? As things that are currently parsing could fail when folks update?

Reply:
Probably so.